src/rsa/padding.rs

```diff
@@ -175,102 +176,100 @@ const PSS_PREFIX_ZEROS: [u8; 8] = [0u8; 8];
 impl Verification for PSS {
     // RSASSA-PSS-VERIFY from https://tools.ietf.org/html/rfc3447#section-8.1.2
     // where steps 1, 2(a), and 2(b) have been done for us.
-    fn verify(&self, msg: untrusted::Input, m: untrusted::Input,
+    fn verify(&self, msg: untrusted::Input, m: &mut untrusted::Reader,
               mod_bits: usize) -> Result<(), error::Unspecified> {
-        m.read_all(error::Unspecified, |m| {
-            // RSASSA-PSS-VERIFY Step 2(c). The `m` this function is given is
-            // the big-endian-encoded value of `m` from the specification,
-            // padded to `k` bytes, where `k` is the length in bytes of the
-            // public modulus. The spec says "Note that emLen will be one less
-            // than k if modBits - 1 is divisible by 8 and equal to k
-            // otherwise," where `k` is the length in octets of the RSA public
-            // modulus `n`. In other words, `em` might have an extra leading
-            // zero byte that we need to strip before we start the PSS decoding
-            // steps which is an artifact of the `Verification` interface.
-            if (mod_bits - 1) % 8 == 0 {
-                if try!(m.read_byte()) != 0 {
-                    return Err(error::Unspecified);
-                }
-            };
-            let em = m;
-            let em_bits = mod_bits - 1;
-            let em_len = (em_bits + 7) / 8;
-            let top_byte_mask = 0xffu8 >> ((8 * em_len) - em_bits);
+        // RSASSA-PSS-VERIFY Step 2(c). The `m` this function is given is the
```

Comment:
Ditto. I'll verify that I didn't actually change the code in any significant way other than reformatting it.